File: Homework/HW8/HW8.py
class HW8:
    def longestPalindrome(self, s: str) -> int:
        charCounts = [0] * 128
        for c in s:
            charCounts[ord(c)] += 1

        length = 0
        for count in charCounts:
            length += count // 2 * 2
            if length % 2 == 0 and count % 2 == 1:
                length += 1

        return length

File: Homework/HW8/HW8.java
package Homework.HW8;

public class HW8 {

    public int longestPalindrome(String s) {
        int[] charCounts = new int[128];
        for (char c : s.toCharArray()) {
            charCounts[c]++;
        }

        int length = 0;
        for (int count : charCounts) {
            length += count / 2 * 2;
            if (length % 2 == 0 && count % 2 == 1) {
                length++;
            }
        }

        return length;
    }
}
